### PR TITLE
RHTAPINST-99: Upstream MinIO Operator

### PR DESCRIPTION
**Pull request by otaviof**
Replacing OpenShift Operator Hub MinIO subscription with their upstream Helm Chart deployment.

### CODE REVIEW

File: installer/charts/rhtap-infrastructure/templates/minio/job-post-deploy.yaml [hunk 1]
```diff
@@ -68,12 +68,15 @@ spec:
...
               mc alias set minio "{{ $minIOURL }}" "${MINIO_USERNAME}" "${MINIO_PASSWORD}"
     {{- range $topic := $v.kafkaNotify.topics }}
-              {{
-                  printf "mc event add minio/%s arn:minio:sqs::%s:kafka --event \"%s\""
-                  $topic.bucket
-                  (upper $topic.bucket)
-                  (join "," $topic.events)
-              }}
+      {{- $event := printf "arn:minio:sqs::%s:kafka" (upper $topic.bucket) }}
+              if [[ -z "$(mc event list minio/{{ $topic.bucket }} {{ $event }})" ]]; then
```
Comment (Roming22) [inline installer/charts/rhtap-infrastructure/templates/minio/job-post-deploy.yaml:72]: Nice, this should fix redeployment failures. We might to have that fix in 1.2.1 as well.

File: installer/charts/rhtap-minio-operator/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+apiVersion: v2
+name: rhtap-minio-operator
+description: Helm Chart to manage the MinIO Operator for RHTAP
+type: application
+version: 0.0.1
+
+dependencies:
+  - name: operator
+    version: 5.0.17
```
Comment (Roming22) [inline installer/charts/rhtap-minio-operator/Chart.yaml:10]: Why not  6.x?

File: installer/charts/rhtap-minio-operator/templates/NOTES.txt
Comment (Roming22) [inline installer/charts/rhtap-minio-operator/templates/NOTES.txt:1]: Why no text at all?

File: installer/charts/rhtap-minio-operator/templates/securitycontextcontraints.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    app: minio-operator
+    {{- include "rhtap-minio-operator.labels" . | nindent 4 }}
+  name: {{ $saAdmin }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
```
Comment (Roming22) [inline installer/charts/rhtap-minio-operator/templates/securitycontextcontraints.yaml:15]: Are such wide permissions really needed?

File: installer/charts/rhtap-minio-operator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  operator:
+    replicaCount: 1
+    additionalLabels:
+      app: minio-operator
+    env:
+      - name: MINIO_OPERATOR_DEPLOYMENT_NAME
+        valueFrom:
+          fieldRef:
+            fieldPath: metadata.labels['app.kubernetes.io/name']
+      - name: MINIO_CONSOLE_TLS_ENABLE
```
Comment (Roming22) [inline installer/charts/rhtap-minio-operator/values.yaml:18]: Shouldn't `MINIO_CONSOLE_TLS_ENABLE` and `OPERATOR_STS_ENABLED` be 'on' by default? (cf
https://min.io/docs/minio/kubernetes/upstream/operations/installation.html#kubernetes-tls-certificate-api)

File: installer/charts/rhtap-minio-operator/Chart.yaml (hunk 2 above)
Comment (otaviof) [reply installer/charts/rhtap-minio-operator/Chart.yaml:10]: 6.x is paywalled.

File: installer/charts/rhtap-minio-operator/templates/NOTES.txt
Comment (otaviof) [reply installer/charts/rhtap-minio-operator/templates/NOTES.txt:1]: We can add it later on.

File: installer/charts/rhtap-minio-operator/templates/securitycontextcontraints.yaml (hunk 3 above)
Comment (otaviof) [reply installer/charts/rhtap-minio-operator/templates/securitycontextcontraints.yaml:15]: Yes. We need to link the SA with the SCC allowing UID/GID 1000.

File: installer/charts/rhtap-minio-operator/values.yaml (hunk 4 above)
Comment (otaviof) [reply installer/charts/rhtap-minio-operator/values.yaml:18]: Console is disabled. STS enabled shows errors into a crash loop back off.

File: installer/charts/rhtap-infrastructure/templates/minio/job-post-deploy.yaml (hunk 1 above)
Comment (otaviof) [reply installer/charts/rhtap-infrastructure/templates/minio/job-post-deploy.yaml:72]: Thanks 🙏 Trying to re-run the installer was causing this job to fail. One think I'd like to improve here is having it as a standalone script, or maybe, consider an `rhtap integration` subcommand -- thus we could run the installer's container image to configure the service.

File: installer/charts/rhtap-minio-operator/Chart.yaml (hunk 2 above)
Comment (otaviof) [reply installer/charts/rhtap-minio-operator/Chart.yaml:10]: And `5.0.17` is the latest supported 5x version, for the record.

**Comment by Roming22** [comment]
@otaviof I propose to wait for the PRs we would want in 1.2.1 before merging that PR.

**Review by Roming22 — APPROVED** [review]
/lgtm

**Comment by otaviof** [comment]
@Roming22, I'm tackling this issue:
```
[install] Error: install failed: Unable to continue with install: ServiceAccount "minio-operator" in namespace "minio-operator" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "rhtap-minio-operator"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "minio-operator
```
By aligning the service-account created by the sub-chart and what our installer expects.

**Comment by Roming22** [comment]
@otaviof Something is broken

```
############################################################
# [3/9] Deploying 'charts/rhtap-minio-operator' in 'minio-operator'.
############################################################
Error: install failed: Unable to continue with install: ServiceAccount "minio-operator" in namespace "minio-operator" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "rhtap-minio-operator"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "minio-operator"
```

**Pull request merged**